src/contexts/useArchaeologistService.ts

```diff
@@ -0,0 +1,93 @@
+import { useState } from 'react';
```

Comment:
I would have comments on about how this is structured but I plan to make a context/reducer with dispatch actions where a lot of this can be moved to.

Reply:
agreed